gossipd/gossip.c

```diff
@@ -1782,11 +1785,27 @@ static struct io_plan *get_peers(struct io_conn *conn,
 
 		id[n] = peer->id;
 		wireaddr[n] = peer->addr;
+
+		struct node* nd = NULL;
+		for (nd = node_map_first(daemon->rstate->nodes, &it); nd; nd = node_map_next(daemon->rstate->nodes, &it)) {
+			if (pubkey_eq(&nd->id, &peer->id)) {
+				append_node(&nodes, nd);
+				break;
+			}
+		}
+		if (nd == NULL) {
+			/* Haven't received node announcement yet */
+			struct node newnode;
+			newnode.id = peer->id;
+			newnode.last_timestamp = -1;
+			newnode.addresses = NULL;
+			append_node(&nodes, &newnode);
```

Comment:
`newnode.alias` is not being loaded here?

Reply:
ugh, for some reason, I thought that we need to create dummy node objects if the node metadata was not available. But it doesn't have to be that way. I will remove this piece of code so that only nodes with full metadata (alias etc.) are sent in the `nodes` array.